datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java

```diff
@@ -219,4 +234,28 @@ private void setupClusterRequest(StackV4Request stackRequest) {
             cluster.setPassword(PasswordUtil.generatePassword());
         }
     }
+
+    void setupCloudStorageAccountMapping(StackV4Request stackRequest, String environmentCrn, IdBrokerMappingSource mappingSource, String cloudPlatform) {
+        CloudStorageRequest cloudStorage = stackRequest.getCluster().getCloudStorage();
+        if (cloudStorage != null && cloudStorage.getAccountMapping() == null) {
+            // In case of SdxClusterRequest with cloud storage, or SdxInternalClusterRequest with cloud storage but missing "accountMapping" property,
+            // getAccountMapping() == null means we need to fetch mappings from IDBMMS.
+            if (mappingSource == IdBrokerMappingSource.IDBMMS) {
+                LOGGER.info("Fetching account mappings from IDBMMS for environment {}", environmentCrn);
+                MappingsConfig mappingsConfig = idbmmsClient.getMappingsConfig(IAM_INTERNAL_ACTOR_CRN, environmentCrn, Optional.empty());
```

Comment:
I'm not sure if the use of Internal actor CRN usage here is correct. 
This setup is happenning in the context of some user. Internal CRN usage should be reserved for API calls that are initiated by services without user interaction. For e.g. Periscope periodically polling a Cloudbreak API.

Reply:
IDBMMS only serves requests to that specific method only for the internal actor, as it is meant to be an internal-only operation. (Request will always fail when using some other actor CRN.) It will not be made available to end users via public APIs (Environments2 REST, CDP CLI). Note that the public APIs do provide a `getMappings` operation that is indeed the user-facing variants, but whose response is inappropriate for us here.
In short, we must use `getMappingsConfig` and that requires the internal actor. 😃

Reply:
Comment has been added to justify the usage of the internal actor.